homeassistant/components/zwave_js/api.py

```diff
@@ -1191,43 +1214,33 @@ async def get(self, request: web.Request, config_entry_id: str) -> web.Response:
         )
 
 
-class NodeDumpView(HomeAssistantView):
-    """View to dump the state of a node on the Z-Wave JS server."""
-
-    url = r"/api/zwave_js/dump/{config_entry_id}/{node_id:\d+}"
-    name = "api:zwave_js:dump:node"
-
-    async def get(
-        self, request: web.Request, config_entry_id: str, node_id: str
-    ) -> web.Response:
-        """Dump the state of a Z-Wave node."""
-        if not request["hass_user"].is_admin:
-            raise Unauthorized()
-        hass = request.app["hass"]
-
-        if config_entry_id not in hass.data[DOMAIN]:
-            raise web_exceptions.HTTPBadRequest
-
-        client: Client = hass.data[DOMAIN][config_entry_id][DATA_CLIENT]
-        node = client.driver.controller.nodes.get(int(node_id))
-        if not node:
-            raise web_exceptions.HTTPNotFound
-
-        version_info = {
-            "driverVersion": client.version.driver_version,
-            "serverVersion": client.version.server_version,
-            "minSchemaVersion": client.version.min_schema_version,
-            "maxSchemaVersion": client.version.max_schema_version,
-        }
-
-        filename = f"zwave_js_node_{node.node_id}_state.json"
-        return web.Response(
-            body=json.dumps([version_info, node.data], indent=2) + "\n",
-            headers={
-                hdrs.CONTENT_TYPE: "application/json",
-                hdrs.CONTENT_DISPOSITION: f'attachment; filename="{filename}"',
-            },
-        )
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/version_info",
+        vol.Required(ENTRY_ID): str,
+    },
+)
+@websocket_api.async_response
+@async_get_entry
+async def websocket_version_info(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict,
+    entry: ConfigEntry,
+    client: Client,
+) -> None:
+    """Get version info from the Z-Wave JS server."""
+    version_info = {
+        "driverVersion": client.version.driver_version,
```

Comment:
Should we use camel case or snake case for the strings?

Reply:
I wasn't sure. If we use automation traces as an example, the keys coming from the backend would be snake and the keys generated in the front end for the download are camel. But the state dump from the backend will be camel, so we are going to have everything be camel but the version info if we switch. But they are also two separate commands, so maybe that's ok? I'm good either way, I spent more time thinking about this than I expected to 🤷‍♂️

Reply:
Let's go with what you prefer.

Reply:
I'll switch to snake since we may use that command for another use and that's the standard for dicts we send